sdk/core/Azure.Core/src/Shared/OperationInternal.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class OperationInternal<TValue, TResult>
+    {
+        private readonly string _operationTypeName;
+
+        private readonly TimeSpan _defaultPollingInterval;
+
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        private readonly Func<CancellationToken, Task<Response<TResult>>> _getResponseAsync;
+
+        private readonly Func<CancellationToken, Response<TResult>> _getResponse;
+
+        private readonly Func<Response<TResult>, OperationInternalResponseStatus> _getStatus;
+
+        private readonly Func<Response<TResult>, TValue> _parseResponse;
+
+        private readonly Func<Response<TResult>, RequestFailedException> _getFailure;
+
+        private Response _rawResponse;
+
+        private TValue _value;
+
+        private RequestFailedException _requestFailedException;
+
+        public OperationInternal(string operationTypeName,
+            ClientDiagnostics clientDiagnostics,
+            Func<CancellationToken, Task<Response<TResult>>> getResponseAsync,
+            Func<CancellationToken, Response<TResult>> getResponse,
+            Func<Response<TResult>, OperationInternalResponseStatus> getStatus,
+            Func<Response<TResult>, TValue> parseResponse,
+            Func<Response<TResult>, RequestFailedException> getFailure,
+            TimeSpan? defaultPollingInterval = null)
```

Comment:
`KeyVault` and `MixedReality` use a different default polling interval, so I'm making it settable here.